pymeasure/instruments/aimtti/aimttiPL.py

```diff
@@ -0,0 +1,176 @@
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+
+
+class PLChannel(Channel):
+    """ A channel of AimTTI PL series power supplies. """
+
+    CURRENT_RANGE = {
+        "LOW": 1,
+        "HIGH": 2
+    }
+
+    def __init__(self, parent, id,
+                 voltage_range: list = None,
+                 current_range: list = None):
+        super().__init__(parent, id)
+        self.voltage_setpoint_values = voltage_range
+        self.current_limit_values = current_range
+
+    voltage_setpoint = Channel.control(
+        "V{ch}?", "V{ch}V %g",
+        """ Control the output voltage of this channel.
+        With verify: the operation is completed when the parameter being adjusted 
+        reaches the required value to within ±5% or ±10 counts.""",
+        validator=strict_range,
+        values=[0, 6],
+        dynamic=True,
+        get_process=lambda x: float(x[3:])
+        )
+
+    current_limit = Channel.control(
+        "I{ch}?", "I{ch} %g",
+        """ A floating point property that controls the current limit in Amps.""",
+        validator=strict_range,
+        values=[0, 1.5],
+        dynamic=True,
+        get_process=lambda x: float(x[3:])
+        )
+
+    voltage = Channel.measurement(
+        "V{ch}O?",
+        """ Read the output readback voltage for this output channel in Volts.""",
+        get_process=lambda x: float(x[:-1])
+        )
+
+    current = Channel.measurement(
+        "I{ch}O?",
+        """ Read the output readback current for this output channel in Amps.""",
+        get_process=lambda x: float(x[:-1])
+        )
+
+    current_range = Channel.control(
+        "IRANGE{ch}?", "IRANGE{ch} %g",
+        """ A string property that sets the current range of the channel. 
+        Low (500/800mA) range, or High range. 
+        Output must be switched off before changing range.""",
+        validator=strict_discrete_set,
+        values=CURRENT_RANGE,
+        map_values=True
+        )
+
+    def enable(self):
+        """ Enables the output."""
+        self.write("OP{ch} 1")
+
+    def disable(self):
+        """ Disables the output."""
+        self.write("OP{ch} 0")
+
+
+class PLBase(Instrument):
+    """ Control AimTTI PL series power supplies. Model number ending with -P or P(G) support this remote interface.
+
+    Documentation: https://resources.aimtti.com/manuals/New_PL+PL-P_Series_Instruction_Manual-Iss18.pdf
+    PL-series devices: https://www.aimtti.com/product-category/dc-power-supplies/aim-plseries
+
+    .. code-block:: python
+
+        psu = PL303MQDP("ASRL7::INSTR")
+        psu.reset()
+        psu.ch_2.voltage = 1.2
+        psu.ch_2.enable()
+
+    """
+
+    chs = Instrument.ChannelCreator(PLChannel, ("1", "2"), voltage_range=[0, 30], current_range=[0, 3])
+    """ Channels of the power supply. The channels are number from right-to-left, starting at 1. 
+    Default values are for the voltage and current range are for the PL303MQD-P."""
```

Comment:
I would not define channels in the base instrument at all, as they seem to be overridden anyway?